AutoRest/AutoRest.Core/Settings.cs

```diff
@@ -146,6 +145,18 @@ public string Header
         public bool AddCredentials { get; set; }
 
         /// <summary>
+        /// If set to true, generate client with a ServiceClientCredentials property and optional constructor parameter. 
+        /// </summary>
+        [SettingsInfo(
+            "If set to a file name all the code will be generated to a single file. Otherwise generated cod")]
+        public string OutputFileName { get; set; }
+
+        /// <summary>
+        /// Gets or sets boolean value indicating if code generation language supports all the code to be generated in a single file.
+        /// </summary>
+        public bool IsSingleFileGenerationSupported { get; set; }
```

Comment:
Why is this on Settings? It would make sense to have this be on the CodeGenerator